### ISSUE TITLE
Cannot use Fandom to create X-Ray

### ISSUE THREAD

**Issue by Nok09** [xxyzz/WordDumb#34]
### Describe the bug

I was able to generate X-Ray and word wise using wikipedia, but when I typed in the Fandom URL in the preference setting and try to run it, the error message occured, and I checked the directory under \AppData\\Roaming\\calibre\\plugins, no folder named "worddumb-fandom" existed.

What's even weirer is that after I delected the Fandom URL in the setting, though there was a cursor inside but I can no longer type in/input any word in that text box, neither can I paste any..
![2022-06-07 214528](https://user-images.githubusercontent.com/107052332/172396045-7d4d420f-d242-4f69-a9d7-e0090aa3de27.png)
.

### System Information

OS: Windows 11
Calibre: 5.43.0 64bit
Python: 3.10.5 64bit
Plugin: 3.21.1

### Error message

```text
Starting job: Generating Word Wise and X-Ray for Gardens of the Moon: Book One of The Malazan Book of the Fallen 
Job: "Generating Word Wise and X-Ray for Gardens of the Moon: Book One of The Malazan Book of the Fallen" failed with error: 
Traceback (most recent call last):
  File "calibre\gui2\threaded_jobs.py", line 82, in start_work
  File "calibre_plugins.worddumb.parse_job", line 108, in do_job
  File "calibre_plugins.worddumb.parse_job", line 180, in create_files
  File "calibre_plugins.worddumb.mediawiki", line 92, in __init__
  File "calibre_plugins.worddumb.mediawiki", line 45, in __init__
  File "pathlib.py", line 1284, in mkdir
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\justi\\AppData\\Roaming\\calibre\\plugins\\worddumb-fandom\\malazan.fandom.com'
 
Called with args: ((26, 'MOBI', 'C:\\Users\\justi\\Calibre Library\\Steven Erikson\\Gardens of the Moon_ Book One of The (26)\\Gardens of the Moon_ Book One o - Steven Erikson.mobi', <calibre.ebooks.metadata.book.base.Metadata object at 0x000002441C105AF0>, {'spacy': 'en_core_web_', 'wiki': 'en'}), True, True) {'notifications': <queue.Queue object at 0x000002441C1057C0>, 'abort': <threading.Event object at 0x000002441C105850>, 'log': <calibre.utils.logging.GUILog object at 0x000002441C105A30>}
```

### Reproduce steps

1. Go to Calibre
2. Click on WordDumb
3. Error message occured

**Comment by xxyzz** [comment]
The error is fixed in https://github.com/xxyzz/WordDumb/commit/50c3b5c3d0d0cb607ff21462cfdd1a0e81e5ceca.

The Fandom URL input box only accepts URLs that conform to the format of the placeholder text, for example: https://malazan.fandom.com or https://malazan.fandom.com/es.

But unfortunately, you still can't use this feature because Fandom disabled the TextExtracts API. I [asked](https://community.fandom.com/wiki/User_blog:Brandon_Rhea/Fandom_is_upgrading_to_MediaWiki_1.37_as_part_of_our_commitment_to_keeping_the_platform_up_to_date?commentId=4400000000003377914) them a few months ago but they haven't responded yet. You could [submit a request](https://support.fandom.com/hc/en-us/requests/new) to Fandom staff or post a comment at [here](https://community.fandom.com/wiki/User_blog:Brandon_Rhea/Fandom_is_upgrading_to_MediaWiki_1.37_as_part_of_our_commitment_to_keeping_the_platform_up_to_date#articleComments) to raise the awareness of this issue.

**Comment by Nok09** [comment]
Thanks a lot, I'll leave a comment there and see how things go. Appreciated.

**Comment by xxyzz** [comment]
https://github.com/xxyzz/WordDumb/commit/85b165ef4f8a5151f201090b02ce811e5f4e4241 uses another API to get data from Fandom, but it's slower than the TextExtracts API.

**Comment by xxyzz** [comment]
The above mentioned commit is included in release v3.27.0.
